### PR TITLE
Build reset url with url module, add option for passing oauth params fro...

### PR DESCRIPTION
**Pull request by cadecairos**
...m id.wmo

### CODE REVIEW

**Comment by cadecairos** [comment]
R?

File: app/db/models/index.js [hunk 1]
```diff
@@ -374,26 +375,33 @@ module.exports = function (sequelize, env) {
...
-    createResetCode: function (user, appURL, callback) {
+    createResetCode: function (user, appURL, oauthParams, callback) {
+
       var code = hat(RESET_CODE_BIT_LENGTH, RESET_CODE_BASE);
       var userResetCode = resetCode.build({
         code: code,
         UserId: user.id
       });
 
+      var resetUrlObj = url.parse(appURL || env.get("WEBMAKERORG"), true);
```
Comment (jbuck) [inline app/db/models/index.js]: Couldn't appURL just contain the querystring parameters needed, instead of having to pass them separately?

**Comment by jbuck** [comment]
@cadecairos I'm wondering if we can simplify this approach here a bit. If that's not possible, then r+

**Comment by jbuck** [comment]
@cadecairos r+ but failing travis. Merge once that goes green?

**Pull request merged**